diff --git a/.gitignore b/.gitignore
index 6e751e203..d9091f1b1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -57,3 +57,5 @@ gaptest_reports/
 # Release script dry-run output
 release_dry_run/
 artifact.txt
+
+.gap9-venv
diff --git a/install_python_deps.sh b/install_python_deps.sh
index cac3e1885..578bb5f71 100755
--- a/install_python_deps.sh
+++ b/install_python_deps.sh
@@ -69,7 +69,7 @@ then
         python -m pip install -r tools/nntool/tests/requirements.txt
     fi
 
-    python -m pip install -r tools/audio-framework/requirements.txt
+    # python -m pip install -r tools/audio-framework/requirements.txt
     python -m pip install -r utils/gapy_v2/requirements.txt
     python -m pip install -r doc/requirements.txt
 
diff --git a/utils/gapy_v2/bin/gapylib/chips/gap/gap9_v2/board_runner.py b/utils/gapy_v2/bin/gapylib/chips/gap/gap9_v2/board_runner.py
index 62ee5cac2..d872073f7 100644
--- a/utils/gapy_v2/bin/gapylib/chips/gap/gap9_v2/board_runner.py
+++ b/utils/gapy_v2/bin/gapylib/chips/gap/gap9_v2/board_runner.py
@@ -342,7 +342,7 @@ class Openocd():
         else:
             raise RuntimeError('Failed to connect to openocd after 30 retries')
 
-        self.telnet = pexpect.spawn(f'telnet localhost {port}', encoding='utf-8', echo=False)
+        self.telnet = pexpect.spawn(f'telnet host.docker.internal {port}', encoding='utf-8', echo=False)
         match = self.telnet.expect(['Open On-Chip Debugger'], timeout=None)
 
     def write(self, addr: int, value: int):
diff --git a/utils/openocd_tools/tcl/gap9revb.tcl b/utils/openocd_tools/tcl/gap9revb.tcl
index 9e67f67c7..697483112 100644
--- a/utils/openocd_tools/tcl/gap9revb.tcl
+++ b/utils/openocd_tools/tcl/gap9revb.tcl
@@ -24,8 +24,8 @@ proc jtag_init {} {
     #gap_reset 0 100
 
     # wait for jtag ready
-    poll_confreg 0x1
-    echo "INIT: confreg polling done"
+    # poll_confreg 0x1
+    # echo "INIT: confreg polling done"
 
     #UNCOMMENT IF YOU UNCOMMENT gap_init 1
     #gap_start 1
@@ -39,8 +39,8 @@ proc init_reset {mode} {
     #targets $::_FC
     gap_reset 1 100
     # wait for jtag ready
-    poll_confreg 0x1
-    echo "RESET: confreg polling done"
+    # poll_confreg 0x1
+    # echo "RESET: confreg polling done"
     jtag arp_init
 }
 
